### PR TITLE
Enable to configure MaxWaitForUnmountDuration

### PR DESCRIPTION
**Pull request by chestack**
enable to configure MaxWaitForUnmountDuration to speed up
volume detach then attach

Closes-Bug: #EAS-21343

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

File: cmd/kube-controller-manager/app/core.go [hunk 1]
```diff
@@ -175,6 +175,13 @@ func startAttachDetachController(ctx ControllerContext) (bool, error) {
 	if ctx.Options.ReconcilerSyncLoopPeriod.Duration < time.Second {
 		return true, fmt.Errorf("Duration time must be greater than one second as set via command line option reconcile-sync-loop-period.")
 	}
+
+	timeConfig := attachdetach.DefaultTimerConfig
+	optionMaxWaitForUnmountDuration := ctx.Options.ReconcilerMaxWaitForUnmountDuration.Duration
+	if optionMaxWaitForUnmountDuration != 0 {
```
Comment (ShuYingya) [inline cmd/kube-controller-manager/app/core.go]: throw an exception here if the value of "optionMaxWaitForUnmountDuration" is less than 1 min?

File: cmd/kube-controller-manager/app/options/options.go [hunk 2]
```diff
@@ -228,6 +228,7 @@ func (s *CMServer) AddFlags(fs *pflag.FlagSet, allControllers []string, disabled
 	fs.Float32Var(&s.UnhealthyZoneThreshold, "unhealthy-zone-threshold", 0.55, "Fraction of Nodes in a zone which needs to be not Ready (minimum 3) for zone to be treated as unhealthy. ")
 	fs.BoolVar(&s.DisableAttachDetachReconcilerSync, "disable-attach-detach-reconcile-sync", false, "Disable volume attach detach reconciler sync. Disabling this may cause volumes to be mismatched with pods. Use wisely.")
 	fs.DurationVar(&s.ReconcilerSyncLoopPeriod.Duration, "attach-detach-reconcile-sync-period", s.ReconcilerSyncLoopPeriod.Duration, "The reconciler sync wait time between volume attach detach. This duration must be larger than one second, and increasing this value from the default may allow for volumes to be mismatched with pods.")
+	fs.DurationVar(&s.ReconcilerMaxWaitForUnmountDuration.Duration, "attach-detach-reconcile-max-wait-unmount-duration", s.ReconcilerMaxWaitForUnmountDuration.Duration, "maximum amount of time the attach detach controller will wait for a volume to be safely unmounted from its node.")
```
Comment (ShuYingya) [inline cmd/kube-controller-manager/app/options/options.go]: I am thinking if it is reasonable to add a symbol on our private options in the description.

File: pkg/apis/componentconfig/types.go [hunk 3]
```diff
@@ -353,6 +353,10 @@ type KubeControllerManagerConfiguration struct {
 	// ReconcilerSyncLoopPeriod is the amount of time the reconciler sync states loop
 	// wait between successive executions. Is set to 5 sec by default.
 	ReconcilerSyncLoopPeriod metav1.Duration
+    // ReconcilerMaxWaitForUnmountDuration is the maximum amount of time the
```
Comment (ShuYingya) [inline pkg/apis/componentconfig/types.go]: please keep the comments aligned

Comment (chestack) [reply pkg/apis/componentconfig/types.go]: fix it in next patch

File: cmd/kube-controller-manager/app/core.go (hunk 1 above)
Comment (chestack) [reply cmd/kube-controller-manager/app/core.go]: good point. Add it in next patch

File: cmd/kube-controller-manager/app/options/options.go (hunk 2 above)
Comment (chestack) [reply cmd/kube-controller-manager/app/options/options.go]: I think it's a common option, we do not need any specific symbol

**Comment by chestack** [comment]
new pr on branch v1.9.8-es

**Pull request closed without merging**